src/Var.cc

```diff
@@ -783,7 +787,11 @@ TraversalCode OuterIDBindingFinder::PreStmt(const Stmt* stmt)
 	auto ws = static_cast<const WhenStmt*>(stmt);
 
 	for ( auto& cl : ws->Info()->WhenExprLocals() )
-		outer_id_references.insert(const_cast<ID*>(cl.get()));
+		{
```

Comment:
Could add a comment here about what this is actually doing? I think I understand that it's only considering this a reference if didn't find the same id in the local scope, but it's not terribly obvious.

Reply:
Thanks for the nudge to revisit that code.  I think you'll find the replacement PR I'm about to submit aesthetically more pleasing ;-).